foyer/AST2NetworkX.py

```diff
@@ -0,0 +1,190 @@
+        # add all edges to the graph
+        for atom in ASTtree.tail:
+            if atom.head == 'atom':
+                # atom is the type that want to add to the graph
+                if atom.is_first_kid and (atom.parent().head == 'branch'):
+                    # if this atom is the first one in its branch then it should connect to the trunk
+                    assert trunk is not None, "can't add branch to a None root!"
+                    self.NetworkX.add_edge(self.get_atom_with_id(atom), self.get_atom_with_id(trunk),
+                                           name=self.get_edge_name(id(atom), id(trunk)))
+                if not atom.is_last_kid:
+                    # if this atom is not the last one, it should connect to the its next atom
+                    if atom.next_kid.head == 'atom':
+                        self.NetworkX.add_edge(self.get_atom_with_id(atom), self.get_atom_with_id(atom.next_kid),
+                                               name=self.get_edge_name(atom, atom.next_kid))
+                    elif atom.next_kid.head == 'branch':
+                        # if the next atom is a new branch then this atom should be the trunk for the new branch
+                        trunk = atom
+                else:
+                    return  # we already travel through the whole branch
+            elif atom.head == 'branch':
+                # a new branch appeared, so we recursively travel to the new branch
+                self.add_edges(atom, trunk)
+
+    def add_label_edges(self):
+        # connect all the atoms with same atom_label together
+        for atom_id, labels in self.atom_with_label.items():
+            for atom_id_inner, labels_inner in self.atom_with_label.items():
+                if atom_id_inner is not atom_id:
+                    for label_inner in labels_inner:
+                        if label_inner in labels:
+                            self.NetworkX.add_edge(self.get_atom_with_id(atom_id_inner), self.get_atom_with_id(atom_id),
+                                                   name=self.get_edge_name(atom_id_inner, atom_id))
+
+    # ===== visualize function ===== #
+    def visualzie(self):
+        # draw the graph without position information by Matplotlib
+        # https://networkx.github.io/documentation/networkx-1.10/reference/generated/networkx.drawing.nx_pylab.draw_networkx.html#networkx.drawing.nx_pylab.draw_networkx
+        plt.figure()
+        nx.draw_networkx(self.NetworkX)
+        plt.show()  # will stuck here unless you close the figure window
+
+    def to_file(self, name_graphml='AST2NX.graphml'):
+        # write to a graphml file which can be read by a lot of professional visualization tools such as Cytoscape
+        if name_graphml.endswith('.graphml'):
+            nx.write_graphml(self.NetworkX, name_graphml)
+        else:
+            nx.write_graphml(self.NetworkX, name_graphml + '.graphml')
+
+    def print_list(self):
```

Comment:
Please remove print_list and print_dict. I assume they are only used for debugging purposes.